product-accelerators/apim/components/org.wso2.healthcare.apim.backendauth/src/main/java/org/wso2/healthcare/apim/backendauth/Utils.java

```diff
@@ -294,4 +295,44 @@ private static void handleTokenRequestError(MessageContext messageContext, int r
         throw exp;
     }
 
+    /**
+     * Evaluate synapse configValue and return the value.
+     * Supported expressions: $header:<header_name>, $ctx:<property_name>
+     *
+     * @param configValue      - configValue to evaluate
+     * @param messageContext  - message context
+     * @return - evaluated value
+     */
+    public static String resolveConfigValues(String configValue, MessageContext messageContext) {
+
+        String headerName = "";
+        if (!configValue.contains("$")) {
+            return configValue;
+        }
+        if (configValue.contains("$ctx")) {
+            return messageContext.getProperty(configValue.substring(configValue.indexOf(":") + 1)).toString();
+        }
+        if (configValue.contains("$header")) {
```

Comment:
Let's have this figured at the deployment of the class mediator. No need to check for each requrest. Have a object to hold the type and value, based on the type you can execute the path